### PR TITLE
add section on mutation within iterations

### PR DESCRIPTION
**Pull request by jackfranklin**
@maxmurdoch @harrison

### CODE REVIEW

**Comment by samjewell** [comment]
:+1: - love how we're getting this in.

Wondering if both the bad and good options should have map? So we're comparing apples to apples? Could also have both map and forEach in the bad section?

File: README.md [hunk 1]
```diff
@@ -203,6 +203,25 @@ function isOdd(x) {
...
+
+// good - don't mutate the array, but create and modify a new one:
+payments = _.cloneDeep(payments).map(function(payment) {
+  payment.id = 'ABC123';
+  return payment;
+});
+```
+
+In practice we may not use `_.cloneDeep` - alter the above code as appropriate based on the libraries available.
+
```
Comment (maxmurdoch) [inline README.md]: :+1:

**Comment by jackfranklin** [comment]
@samjewell +1, added that in

**Comment by harrison** [comment]
:+1:

**Pull request merged**